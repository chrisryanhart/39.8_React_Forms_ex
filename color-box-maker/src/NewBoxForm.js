import React, {useState} from 'react';

const NewBoxForm = ({addBox}) => {
    const INITIAL_STATE = {width: "", height: "", backgroundColor: ""}

    const [boxFormData, setBoxFormData] = useState(INITIAL_STATE);


    const handleChange = (e) => {
        const { name, value } = e.target;
        setBoxFormData(data => ({
            ...data,
            [name]: value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        const { width, height, backgroundColor } = boxFormData;
        // setItem
        addBox(width, height, backgroundColor);
        setBoxFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <div>
                <label htmlFor='width'>Width: </label>
                <input id='width' name='width' value={boxFormData.width} onChange={handleChange}/>

                <label htmlFor='height'>Height: </label>
                <input 
                    id='height' 
                    name='height'
                    value={boxFormData.height}
                    onChange={handleChange}
                />

                <label htmlFor='backgroundColor'>BG Color: </label>
                <input 
                    id='backgroundColor' 
                    name='backgroundColor' 
                    value={boxFormData.backgroundColor}
                    onChange={handleChange}
                />
            </div>
            <button>Add Box</button>
        </form>
    );
}

export default NewBoxForm;