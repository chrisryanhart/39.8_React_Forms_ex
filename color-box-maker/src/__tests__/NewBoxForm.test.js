import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from '../NewBoxForm';

it("renders without crashing", function() {
    render(<NewBoxForm/>);
})

it("renders without crashing", function() {
    const {asFragment} = render(<NewBoxForm/>);
    expect(asFragment()).toMatchSnapshot();
})


// add box function works
it("runs the addBox function on form submit", function() {
    const addBoxMock = jest.fn();
    const { getByText } = render(<NewBoxForm addBox={addBoxMock} />);
    const addBoxButton = getByText("Add Box");
    fireEvent.click(addBoxButton);
    expect(addBoxMock).toHaveBeenCalled();
  });