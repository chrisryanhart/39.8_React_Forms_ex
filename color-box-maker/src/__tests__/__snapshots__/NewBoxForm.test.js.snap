// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        name="width"
        value=""
      />
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        id="height"
        name="height"
        value=""
      />
      <label
        for="backgroundColor"
      >
        BG Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        value=""
      />
    </div>
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
